src/sync/sensor/processors/sensor.processor.ts

```diff
@@ -11,15 +11,55 @@ import {
   SensorRelocated,
   SensorUpdated,
 } from 'src/events/sensor';
+import {LedgerConnection} from "../ledger.connection";
+const { Gateway, Wallets } = require('fabric-network');
+const path = require('path');
+const fs = require('fs');
 
 @Injectable()
 export class SensorProcessor {
   protected logger: Logger = new Logger(this.constructor.name);
 
+  constructor(private readonly ledgerConnection: LedgerConnection) {
+  }
+
   async process(event): Promise<void> {
+    if (event instanceof SensorRegistered) {
+      await this.processCreated(event);
+    } else if (event instanceof SensorUpdated) {
+      await this.processUpdated(event);
+    } else if (event instanceof SensorDeleted) {
+      await this.processDeleted(event);
+    } else if (event instanceof SensorActivated) {
+      await this.processActivated(event);
+    } else if (event instanceof SensorDeactivated) {
+      await this.processDeactivated(event);
+    } else if (event instanceof SensorOwnershipShared) {
+      await this.processOwnershipShared(event);
+    } else if (event instanceof SensorOwnershipTransferred) {
+      await this.processOwnershipTransferred(event);
+    } else if (event instanceof DatastreamAdded) {
+      await this.processDataStreamCreated(event);
+    } else if (event instanceof DatastreamDeleted) {
+      await this.processDataStreamDeleted(event);
+    } else if (event instanceof SensorRelocated) {
+      await this.processLocationUpdated(event);
+    } else {
+      this.logger.warn(`Caught unsupported event: ${event}`);
+    }
   }
 
   async processCreated(event: SensorRegistered) {
+    try {
```

Comment:
Dit blok gaan we nog beter maken, niet? Dit is de interface tussen de bridge en de chaincode en elke interactie (submitTransaction) zit binnen hetzelfde block; dat zou prima als Promise beschreven kunnen zijn? Een generieke `await submitEvent(event)` function call die dit block uitvoert voor elk event?

Reply:
Klinkt goed! Laat het maar weten als je die functie aan de chain-code kant af hebt, dan integreer ik m.